conda_auth/handlers/basic_auth.py

```diff
@@ -0,0 +1,104 @@
+"""
+Basic auth implementation for the conda auth handler plugin hook
+"""
+from __future__ import annotations
+
+from getpass import getpass
+
+import keyring
+from keyring.errors import PasswordDeleteError
+from requests.auth import _basic_auth_str  # type: ignore
+from conda.exceptions import CondaError
+from conda.models.channel import Channel
+
+from ..constants import HTTP_BASIC_AUTH_NAME, LOGOUT_ERROR_MESSAGE
+from ..exceptions import CondaAuthError
+from .base import (
+    AuthManager,
+    CacheChannelAuthBase,
+    test_credentials,
+    save_credentials,
+)
+
+USERNAME_PARAM_NAME = "username"
+"""
+Setting name that appears in ``context.channel_settings``; This value is optionally
+set. If not set, we ask for it via the ``input`` function.
+"""
+
+
+class BasicAuthManager(AuthManager):
+    def get_keyring_id(self, channel_name: str):
+        return f"{HTTP_BASIC_AUTH_NAME}::{channel_name}"
+
+    @save_credentials
+    @test_credentials
+    def set_secrets(self, channel: Channel, settings: dict[str, str]) -> None:
+        if self.cache.get(channel.canonical_name) is not None:
+            return
+
+        username = settings.get(USERNAME_PARAM_NAME)
+        keyring_id = self.get_keyring_id(channel.canonical_name)
+
+        if username is None:
+            print(f"Please provide credentials for channel: {channel.canonical_name}")
+            username = input("Username: ")
```

Comment:
It might be good to abstract that prompt into a method, so we can override it later to integrate into Navigator or TUI etc that don't have TTY interaction

Reply:
That makes sense. I always feel a little weird using `input` directly in Python.